core/com.b2international.snowowl.core.tests/src/com/b2international/snowowl/core/uri/ComponentURITest.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2020 B2i Healthcare Pte Ltd, http://b2i.sg
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.b2international.snowowl.core.uri;
+
+import static org.junit.Assert.assertEquals;
+
+import org.junit.Test;
+
+/**
+ * @since 7.12.0
+ */
+public class ComponentURITest {
```

Comment:
It would be great to have a few parse fail tests as well to verify the following:
* Number of parts >= 3 (if not fail with reasonable error message)
* Last part is not empty (if not fail with reasonable error message)
* Second last part is not empty and can be parsed into a short number (if not fail with reasonable error message)